### PR TITLE
Refine the next cycle state refresh function

### PR DESCRIPTION
**Pull request by BowTiedRadone**
This PR updates the state refreshing function for new cycles and targets `feat/pox-4-stateful-property-testing` (#4550).

### CODE REVIEW

File: contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts [hunk 1]
```diff
@@ -64,37 +64,83 @@ export class Stub {
...
+        // Remove the expired stackers from the operator's lockedAddresses list.
         expiredStackers.forEach((expStacker) => {
-          const expStackerWallet = this.stackers.get(expStacker)!;
           const expStackerIndex = wallet.lockedAddresses.indexOf(expStacker);
           wallet.lockedAddresses.splice(expStackerIndex, 1);
+        });
+
+        // For each pool stacker that no longer have partially commited STX for
+        // the next reward cycle, decrement the operator's amountToCommit
+        // (partial-stacked) by the stacker's amountLocked
```
Comment (moodmosaic) [inline contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts:120]:
```suggestion
        // (partial-stacked) by the stacker's amountLocked.
```

File: contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts [hunk 2]
```diff
@@ -64,37 +64,83 @@ export class Stub {
...
-          wallet.unlockHeight > 0 && wallet.unlockHeight <= burnBlockHeight + 1
+          wallet.unlockHeight > 0 && wallet.unlockHeight <= burnBlockHeight
         ) {
           wallet.isStacking = false;
+          wallet.isStackingSolo = false;
           wallet.amountUnlocked += wallet.amountLocked;
           wallet.amountLocked = 0;
           wallet.unlockHeight = 0;
           wallet.firstLockedRewardCycle = 0;
+        } // If the wallet is solo stacking and its stack won't expire in the
```
Comment (moodmosaic) [inline contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts:136]:
```suggestion
        } 
        // If the wallet is solo stacking and its stack won't expire in the
```

**Review by moodmosaic — APPROVED** [review]
LGTM!

File: contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts [hunk 3]
```diff
@@ -64,37 +64,84 @@ export class Stub {
...
           wallet.amountLocked = 0;
           wallet.unlockHeight = 0;
           wallet.firstLockedRewardCycle = 0;
+        } 
+        // If the wallet is solo stacking and its stack won't expire in the
+        // next reward cycle, increment the model's nextRewardSetIndex (the
+        // next empty reward slot)
+        else if (
+          wallet.unlockHeight > 0 &&
+          wallet.unlockHeight > burnBlockHeight + 1050 && wallet.isStackingSolo
```
Comment (friedger) [inline contrib/core-contract-tests/tests/pox-4/pox_CommandModel.ts:142]: nit: 1050 could be a constant.

**Review by friedger — APPROVED** [review]
LGTM

**Pull request merged**